autogpt_platform/backend/backend/blocks/google/gmail.py

```diff
@@ -0,0 +1,444 @@
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import SchemaField
+
+from ._auth import (
+    GOOGLE_OAUTH_IS_CONFIGURED,
+    TEST_CREDENTIALS,
+    TEST_CREDENTIALS_INPUT,
+    GoogleCredentials,
+    GoogleCredentialsField,
+    GoogleCredentialsInput,
+)
+
+
+class GmailReadBlock(Block):
+    class Input(BlockSchema):
+        credentials: GoogleCredentialsInput = GoogleCredentialsField(
+            ["https://www.googleapis.com/auth/gmail.readonly"]
+        )
+        query: str = SchemaField(
+            description="Search query for reading emails",
+            default="is:unread",
+        )
+        max_results: int = SchemaField(
+            description="Maximum number of emails to retrieve",
+            default=10,
+        )
+
+    class Output(BlockSchema):
+        result: list[dict] = SchemaField(
+            description="List of email data",
+        )
```

Comment:
As discussed (internally), outputting a *stream* of items should be the default, outputting a *list* with those items can be added in parallel.

Reply:
For future UX development it would also help a lot if the type was better defined than `dict`

Reply:
done